feat(as-5748): add final comments page

james-hill-kc opened the pull request:
## Ticket Number

https://pins-ds.atlassian.net/browse/AS-5748

## Description of change

<!-- Please describe the change -->

## Checklist

<!-- Put an `x` in all the boxes that apply: -->

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

edpackard-work commented on packages/forms-web-app/src/validators/final-comment/final-comment.js:
```diff
@@ -0,0 +1,25 @@
+} = require('../common/does-not-include-sensitive-information');
+const { numberToLocaleString } = require('../../lib/number-functions');
+
+const maxCharacters = 32500;
```
could/should this go in config file?

james-hill-kc replied:
Just checking, would this be best placed in `src/configuration/config.js`?  And if so should we be pulling the value from an environment variable?

edpackard-work replied:
you could put it there, and do a `key: environment variable || default value` sort of thing, with the default being 32500?

james-hill-kc replied:
Done!

guiribeirokc commented:
Where are the tests at least for the router and the number function created?

james-hill-kc commented:
@guiribeirokc good shout on the unit test for the number-functions method, I've added this now. I also added a test for the route, although as mentioned I think there's room for debate around the value of unit testing routes/controllers etc. But this is a discussion for another time.

edpackard-work commented on packages/forms-web-app/src/views/final-comment/final-comment.njk:
```diff
@@ -0,0 +1,72 @@
+				<h1 class="govuk-heading-l">
+					What is your final comment?
+				</h1>
+        <p>Your final comment should only respond to the statements from the ((local planning authority / appellant)) or interested parties, and must not include any new evidence.</p>
```
confirm with design what is required here?

james-hill-kc replied:
I've confirmed with design, the final comments page should use the "local planning authority" text. The "appellant" text is an alternative version intended for future use on the "LPA" journey. I've updated the template accordingly.

edpackard-work commented on packages/forms-web-app/src/views/final-comment/final-comment.njk:
```diff
@@ -0,0 +1,72 @@
+{% from "govuk/components/checkboxes/macro.njk" import govukCheckboxes %}
+{% from "govuk/components/button/macro.njk" import govukButton %}
+
+{% set title = "Your final comments - Submit final comments - Appeal a planning decision - GOV.UK" %}
```
should this be Submit final comment

james-hill-kc replied:
Updated (also updated the validation error message from "final comments" to "final comment"

james-hill-kc commented on packages/forms-web-app/src/controllers/final-comment/final-comment.js:
```diff
@@ -0,0 +1,55 @@
+		req.session.finalComment.finalComment = finalComment;
+		req.session.finalComment.doesNotIncludeSensitiveInformation = !hasSensitiveInformation;
+
+		return res.redirect(`/full-appeal/submit-final-comments/documents-check`);
```
Thanks, I've updated this now
